frontend/packages/console-dynamic-plugin-sdk/src/app/k8s/api-discovery/api-discovery.ts

```diff
@@ -11,7 +11,7 @@ import { cacheResources, getCachedResources } from './discovery-cache';
 
 const POLLs = {};
 const apiDiscovery = 'apiDiscovery';
-const API_DISCOVERY_POLL_INTERVAL = 60_000;
+const API_DISCOVERY_POLL_INTERVAL = 60_000 * 5;
```

Comment:
I'm a little unsure why running API discovery would cause the browser to refresh at all. Do we know?

This would make it happen less often, but the risk of losing form data is still there.

Reply:
I think the redux state is being updated on every iteration of API discovery (specifically <RootState>.k8s.RESOURCES.models) and that causes dependent components to re-render. This definitely doesn't eliminate the bug but should reduce occurrences. I've thought of other solutions (including getting rid of the poll altogether), but most require more extensive refactors which have higher regression risks.

Reply:
Re-render shouldn't lose form state unless a component is getting unmounted. We might have a bug where a component is getting unmounted/remounted unexpectedly?

Reply:
I'll dig a little bit more.

Reply:
My guess is that the form is actually unmounting and remounting when the k8s resources "inFlight" state gets cycled.